keps/sig-node/runtime-class-scheduling.md

```diff
@@ -190,47 +186,64 @@ inculde:
   RuntimeClasses would need a toleration to enable them to be run on those
   nodes.
 
-### RuntimeController
-
-RuntimeController is a new in-tree admission plugin that should eventually be
-enabled on almost every Kubernetes clusters. The role of the controller for
-scheduling is to merge the topology constraints from the RuntimeClass into the
-PodSpec. Eventually, the controller's responsibilities may grow, such as to
-merge in [pod overhead][] or validate feature compatibility.
-
-Note that the RuntimeController is not needed if we implement [native scheduler
-support](#runtimeclass-aware-scheduling).
-
-#### Mix-in
+### Scheduler
 
-The RuntimeClass topology is merged with the pod's NodeSelector & Tolerations.
+A new scheduler predicate will manage the RuntimeClass topology. It will lookup
+the RuntimeClass associated with the pod being scheduled. If there is no
+RuntimeClass, or the RuntimeClass does not include a topology, then the
+predicate will permit the pod to be scheduled to the evaluated node. Otherwise,
+it will check whether the NodeSelector matches the node.
 
-**NodeSelectorRequirements**
+Adding a dedicated RuntimeClass predicate rather than mixing the rules in to the
+NodeAffinity evaluation means that in the event a pod is unschedulable there
+will be a clear explanation of why. For example:
 
-NodeSelectorRequirements are merged into the pod's node affinity scheduling requirements:
 ```
-pod.spec.affinity.nodeAffinity.requiredDuringSchedulingIgnoredDuringExecution[*].matchExpressions
+0/10 Nodes are available: 5 nodes do not have enough memory, 5 nodes don't match RuntimeClass
 ```
 
-Since the `requiredDuringSchedulingIgnoredDuringExecution` NodeSelectorTerms are
-unioned (OR'd), intersecting the RuntimeClass's NodeSelectorRequirements means
-the requirements need to be appended to _every_ NodeSelectorTerm.
+If the pod's referenced RuntimeClass does not exist at scheduling time, the
+RuntimeClass predicate will fail. The scheduler will periodically retry, and
+once the RuntimeClass is (re)created, the pod will be scheduled.
 
-**Tolerations**
+### RuntimeClass Admission Controller
 
-Merging tolerations is much simpler as we want to _union_ the RuntimeClass
+The RuntimeClass admission controller is a new in-tree admission plugin that
```

Comment:
if the feature gate for runtime class is on, this will always be in the default set right?

Reply:
Yes. Fixed.